### PR TITLE
add descriptions for all argonaut and dstu2 openAPI

### PR DESCRIPTION
**Pull request by eclendenning-va**
Story : https://vasdvp.atlassian.net/browse/SHANK-154

- added descriptions for existing parameters
- corrected incorrect parameters
- created a list of unlisted parameters as a comment in the story

### CODE REVIEW

File: argonaut/src/main/java/gov/va/api/health/argonaut/api/AllergyIntoleranceApi.java [hunk 1]
```diff
@@ -49,7 +49,15 @@
...
   )
   AllergyIntolerance allergyIntoleranceRead(
-      @Parameter(in = ParameterIn.PATH, name = "id", required = true) String id);
+      @Parameter(
+            in = ParameterIn.PATH,
+            name = "id",
+            required = true,
+            description =
+                "The logical id of the resource, as used in the URL for the resource."
+                    + " Once assigned, this value never changes."
```
Comment (jhulbert-va) [inline argonaut/src/main/java/gov/va/api/health/argonaut/api/AllergyIntoleranceApi.java]: Should we maybe substitute out the first 'resource' for the actual resource name (i.e. AllergyIntolerance) so it doesn't sound as repetitive?

Comment (bschofield-va) [reply argonaut/src/main/java/gov/va/api/health/argonaut/api/AllergyIntoleranceApi.java]: _as used in the URL for the resource_ ... this is documenting the URL, so may be we should drop that phrase?

> The logical ID of the resource. Once assigned, this value never changes.

Thoughts?

File: argonaut/src/main/java/gov/va/api/health/argonaut/api/MedicationOrderApi.java [hunk 2]
```diff
@@ -87,7 +95,30 @@ MedicationOrder medicationOrderRead(
...
-      @Parameter(in = ParameterIn.QUERY, required = true, name = "patient") String id,
-      @Parameter(in = ParameterIn.QUERY, name = "page") @DefaultValue("1") int page,
-      @Parameter(in = ParameterIn.QUERY, name = "_count") @DefaultValue("15") int count);
+      @Parameter(
+            in = ParameterIn.QUERY,
+            required = true,
+            name = "patient",
+            description =
+                "Integration Control Number (ICN) assigned by the Master Veteran Index (MVI)"
+                    + " as a link to a resource representing the person"
```
Comment (bschofield-va) [inline argonaut/src/main/java/gov/va/api/health/argonaut/api/MedicationOrderApi.java]: Maybe remove _as a link resource representing_?

> ... (MVI) of the person to whom the medication will be given

Thoughts?

File: argonaut/src/main/java/gov/va/api/health/argonaut/api/ObservationApi.java [hunk 3]
```diff
@@ -87,7 +95,31 @@ Observation observationRead(
...
   Observation.Bundle observationSearch(
-      @Parameter(in = ParameterIn.QUERY, required = true, name = "patient") String id,
-      @Parameter(in = ParameterIn.QUERY, name = "page") @DefaultValue("1") int page,
-      @Parameter(in = ParameterIn.QUERY, name = "_count") @DefaultValue("15") int count);
+      @Parameter(
+            in = ParameterIn.QUERY,
+            required = true,
+            name = "patient",
+            description =
+                "The Integration Control Number (ICN) assigned by the Master Veteran Index (MVI)"
```
Comment (bschofield-va) [inline argonaut/src/main/java/gov/va/api/health/argonaut/api/ObservationApi.java]: zoinks .. that's mouthful. Anyone got any ideas how to make this more concise?

Comment (eclendenning-va) [reply argonaut/src/main/java/gov/va/api/health/argonaut/api/ObservationApi.java]: "The Integration Control Number (ICN) assigned by the Master Veteran Index (MVI) that refers to the patient that is described by the observation"

Does this cut too much?

File: argonaut/src/main/java/gov/va/api/health/argonaut/api/AllergyIntoleranceApi.java (hunk 1 above)
Comment (eclendenning-va) [reply argonaut/src/main/java/gov/va/api/health/argonaut/api/AllergyIntoleranceApi.java]: Thoughts on removing "for the resource" so the id definition stays generic?

Comment (bschofield-va) [reply argonaut/src/main/java/gov/va/api/health/argonaut/api/AllergyIntoleranceApi.java]: > The logical id of the resource. Once assigned, this value never changes.
?

**Pull request merged**